contracts/multisigs/GuardCM.sol

```diff
@@ -0,0 +1,292 @@
+error NotDefeated(uint256 proposalId, ProposalState state);
+
+/// @title GuardCM - Smart contract for Gnosis Safe community multisig (CM) guard
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+contract GuardCM {
+    event GovernorUpdated(address indexed governor);
+    event SetTargetSelectors(address[] indexed targets, bytes4[] indexed selectors, bool[] statuses);
+    event GovernorCheckProposalIdChanged(uint256 indexed proposalId);
+    event GuardPaused(address indexed account);
+    event GuardUnpaused();
+
+    // Owner address
+    address public immutable owner;
+    // Multisig address
+    address public immutable multisig;
+    // Governor address
+    address public governor;
+    // Guard pausing possibility
+    uint8 public paused = 1;
+    
+    // Mapping of address + bytes4 selector => enabled / disabled
+    mapping(uint256 => bool) public mapAllowedTargetSelectors;
+
+    // Schedule selector
+    bytes4 public constant SCHEDULE = bytes4(keccak256(bytes("schedule(address,uint256,bytes,bytes32,bytes32,uint256)")));
+    // ScheduleBatch selector
+    bytes4 public constant SCHEDULE_BATCH = bytes4(keccak256(bytes("scheduleBatch(address[],uint256[],bytes[],bytes32,bytes32,uint256)")));
+    // Initial check governance proposal Id
+    // Calculated from the proposalHash function of the GovernorOLAS with the following parameters:
+    // targets = [address(0)], values = [0], calldatas = [0x], description = "Is governance alive?"
+    uint256 public governorCheckProposalId = 62151151991217526951504761219057817227643973118811130641152828658327965685127;
+
+    /// @dev GuardCM constructor.
+    /// @param _timelock Timelock address.
+    /// @param _multisig Community multisig address.
+    /// @param _governor Governor address.
+    constructor(address _timelock, address _multisig, address _governor) {
+        owner = _timelock;
+        multisig = _multisig;
+        governor = _governor;
+    }
+
+    /// @dev Changes the governor.
+    /// @param newGovernor Address of a new governor.
+    function changeGovernor(address newGovernor) external {
+        if (msg.sender != owner) {
+            revert OwnerOnly(msg.sender, owner);
+        }
+
+        // Check for the zero address
+        if (newGovernor == address(0)) {
+            revert ZeroAddress();
+        }
+
+        governor = newGovernor;
+        emit GovernorUpdated(newGovernor);
+    }
+
+    /// @dev Changes the governor check proposal Id.
+    /// @param proposalId Governor check proposal Id.
+    function changeGovernorCheckProposalId(uint256 proposalId) external {
+        if (msg.sender != owner) {
+            revert OwnerOnly(msg.sender, owner);
+        }
+
+        // Check for the zero value
+        if (proposalId == 0) {
+            revert ZeroValue();
+        }
+
+        governorCheckProposalId = proposalId;
+        emit GovernorCheckProposalIdChanged(proposalId);
```

Comment:
can we also somehow emit also the proposal description?

Reply:
This is a one-way hash.
If the description is understood as text, then it is also included as a hash.